lib/core/include/scipp/core/spatial_transforms.h

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2021 Scipp contributors (https://github.com/scipp)
+/// @file
+/// @author Simon Heybrock
+#pragma once
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+#include "scipp/core/dtype.h"
+
+#include <type_traits>
+#include <optional>
+
+namespace scipp::core {
+
+class Rotation {
+private:
+  // Store as quaterniond as this is more space efficient than storing as matrix
+  // (4 doubles for quat vs 9 doubles for 3x3 matrix).
+  Eigen::Quaterniond m_quat;
+
+public:
+  EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+  Rotation() : m_quat(Eigen::Quaterniond::Identity()){};
+  explicit Rotation(const Eigen::Quaterniond &x) : m_quat(x){};
+
+  [[nodiscard]] Eigen::Quaterniond quat() const {
+    return m_quat;
+  }
+
+  bool operator==(const Rotation &other) const {
+    return m_quat.w() == other.m_quat.w() && 
+        m_quat.x() == other.m_quat.x() && 
+        m_quat.y() == other.m_quat.y() && 
+        m_quat.z() == other.m_quat.z();
+  }
+
+  const double get(const int index) const {
+      if (index == 0)
+          return m_quat.w();
+      else if (index == 1)
+          return m_quat.x();
+      else if (index == 2)
+          return m_quat.y();
+      else if (index == 3)
+          return m_quat.z();
+      else
+          throw std::out_of_range("invalid index in get for Rotation");
+  }
```

Comment:
Fixed the order. 

It's required by `get` in `variable.tcc`. You mentioned in a previous comment that we should just expose the individual coefficients to that function, not the whole eigen object.

Reply:
Are you sure it is getting used? Because:

- In `variable.tcc` we require access with a compile-time index, not run-time as in your implementation.
- The `get()` above returns by value, but in `variable.tcc` we need to get the address of the underlying element, i.e., we need a reference.